src/main/java/net/neoforged/moddevgradle/internal/ModDevPlugin.java

```diff
@@ -207,10 +207,16 @@ public void apply(Project project) {
             task.getParchmentData().from(parchmentData);
 
             var minecraftArtifactsDir = modDevBuildDir.map(dir -> dir.dir("artifacts"));
-            task.getCompiledArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local.jar")));
-            task.getCompiledWithSourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-merged.jar")));
-            task.getSourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-sources.jar")));
-            task.getResourcesArtifact().set(minecraftArtifactsDir.map(dir -> dir.file("neoforge-minecraft-joined-local-resources-aka-client-extra.jar")));
+            Function<String, Provider<RegularFile>> jarPathFactory = suffix -> {
+                return minecraftArtifactsDir.zip(
+                        // It's helpful to be able to differentiate the Vanilla jar and the NeoForge jar in classic multiloader setups.
+                        extension.getVersion().map(v -> "neoforge-" + v).orElse(extension.getNeoFormVersion().map(v -> "vanilla-" + v)),
+                        (dir, prefix) -> dir.file(prefix + "-minecraft" + suffix + ".jar"));
+            };
+            task.getCompiledArtifact().set(jarPathFactory.apply(""));
+            task.getCompiledWithSourcesArtifact().set(jarPathFactory.apply("-merged"));
+            task.getSourcesArtifact().set(jarPathFactory.apply("-sources"));
+            task.getResourcesArtifact().set(jarPathFactory.apply("-resources-aka-client-extra"));
```

Comment:
The client extra suffix will basically always be trimmed, it should be moved up at the start.

Reply:
`-resources` is clearer for users. The only reason we have `client-extra` is because of the ignore list.